### PR TITLE
Init dex_mantle_base_trades

### PR DESCRIPTION
**Pull request by Hosuke**
Trying to add dex_mantle.base_trades

### Contribution type
Please check the type of contribution this pull request is for:

- [X] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

**Comment by jeff-dude** [comment]
for tracking and conversation tomorrow:

> Dexes Mantle would like to add to DEX trades: 
> - Agni (to be added - there was an issue with the contract)
> - Butter
> - fusionx
> - merchant moe
> - Izumi

> Agni, FusionX, and Butter are all forks of either Uniswap v2 or v3. except that Merchant Moe is Trader Joe v2 fork. FYI: https://defillama.com/protocol/merchant-moe-liquidity-book#information.

**Comment by jeff-dude** [comment]
we should also add to [trusted tokens](https://github.com/duneanalytics/spellbook/blob/main/models/prices/prices_trusted_tokens.sql), then maybe have a separate PR to add other recent chains to keep it up-to-date

File: sources/_sector/dex/trades/mantle/_sources.yml
Comment (Hosuke) [inline sources/_sector/dex/trades/mantle/_sources.yml:1]: Currently most swap event tables are still missing. I will refresh regularly.

Comment (jeff-dude) [reply sources/_sector/dex/trades/mantle/_sources.yml:1]: we've raised this to the blockchain team, waiting to hear back

**Comment by Hosuke** [comment]
Sample merchant_moe_mantle_base_trades:
https://dune.com/queries/3926315

**Comment by jeff-dude** [comment]
did we ever add trusted tokens prices for mantle?

then the other DEXs expected for mantle (all supposed to be uniswap v2 or v3 forks):
- agni
- fusionX
- butter
from what i see in data explorer, we have event decoded tables for the pools created, but we don't have swap events for any of these. do you agree?

**Comment by Hosuke** [comment]
> from what i see in data explorer, we have event decoded tables for the pools created, but we don't have swap events for any of these. do you agree?

* agni
-- missing in the decoded project list.
* fusionX
-- missing general Pair_evt_swap decoding
* butter
-- missing general Pair_evt_swap decoding

Also mantle trusted tokens have been added here:
https://github.com/duneanalytics/spellbook/blob/main/models/prices/prices_trusted_tokens.sql#L77

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**